frontend/src/components/panels/Document.vue

```diff
@@ -379,19 +379,29 @@ export default defineComponent({
 			this.canvasCursor = updateMouseCursor.cursor;
 		});
 		this.editor.dispatcher.subscribeJsMessage(TriggerTextCommit, () => {
-			if (this.textInput) this.editor.instance.on_change_text(this.textInput.textContent || "");
+			if (this.textInput) {
+				let text = this.textInput.innerText;
+				if (text[text.length - 1] === "\n") text = text.slice(0, -1);
+
+				this.editor.instance.on_change_text(text);
+			}
 		});
 
 		this.editor.dispatcher.subscribeJsMessage(DisplayEditableTextbox, (displayEditableTextbox) => {
 			this.textInput = document.createElement("DIV") as HTMLDivElement;
 			this.textInput.id = "editable-textbox";
-			this.textInput.textContent = displayEditableTextbox.text;
+			this.textInput.innerText = `${displayEditableTextbox.text}\n`;
 			this.textInput.contentEditable = "true";
 			this.textInput.style.width = displayEditableTextbox.line_width ? `${displayEditableTextbox.line_width}px` : "max-content";
 			this.textInput.style.height = "auto";
 			this.textInput.style.fontSize = `${displayEditableTextbox.font_size}px`;
 			this.textInput.oninput = (): void => {
-				if (this.textInput) this.editor.instance.update_bounds(this.textInput.textContent || "");
+				if (this.textInput) {
+					let text = this.textInput.innerText;
+					if (text[text.length - 1] === "\n") text = text.slice(0, -1);
```

Comment:
This seems to be duplicate code as above. Can you break this out into a function?

Reply:
The shared code is 2 lines used three times (one time has a different editor function) I'm not really sure it is worth sharing them?

Reply:
I'd normally agree with your thoughts that they are too minimal to share, but these two lines are relatively high-entropy and error-prone when doing maintenance on them. Like they could get easily out of sync with one another by accident when maintaining this code someday. So for that reason I feel like a function would be better in this case, even though most average triple-duplicate pairs of lines aren't worth it.